documentation/examples/remote_storage/remote_storage_adapter/main.go

```diff
@@ -217,20 +217,8 @@ func serve(addr string, writers []writer, readers []reader) error {
 	})
 
 	http.HandleFunc("/read", func(w http.ResponseWriter, r *http.Request) {
-		compressed, err := ioutil.ReadAll(r.Body)
+		req, err := remote.DecodeRequest(r)
```

Comment:
The dependencies are not the same. In v1 they merely happen to be in the same directory, what you're doing here goes further and makes that an actual code level dependency.